data-prepper-plugins/opensearch/src/main/java/org/opensearch/dataprepper/plugins/sink/opensearch/index/IndexConfiguration.java

```diff
@@ -188,133 +178,93 @@ private void determineTemplateType(Builder builder) {
+                .withNumReplicas(openSearchSinkConfig.getNumReplicas())
+                .withBulkSize(openSearchSinkConfig.getBulkSize(DEFAULT_BULK_SIZE))
+                .withEstimateBulkSizeUsingCompression(openSearchSinkConfig.getEstimateBulkSizeUsingCompression(DEFAULT_ESTIMATE_BULK_SIZE_USING_COMPRESSION))
+                .withMaxLocalCompressionsForEstimation(openSearchSinkConfig.getMaxLocalCompressionsForEstimation(DEFAULT_MAX_LOCAL_COMPRESSIONS_FOR_ESTIMATION))
+                .withFlushTimeout(openSearchSinkConfig.getFlushTimeout(DEFAULT_FLUSH_TIMEOUT))
+                .withVersionType(openSearchSinkConfig.getVersionType())
+                .withNormalizeIndex(openSearchSinkConfig.getNormalizeIndex())
+                .withIsmPolicyFile(openSearchSinkConfig.getIsmPolicyFile())
+                .withDocumentRootKey(openSearchSinkConfig.getDocumentRootKey())
+                .withDistributionVersion(openSearchSinkConfig.getDistributionVersion());
 
-        final String versionExpression = pluginSetting.getStringOrDefault(DOCUMENT_VERSION_EXPRESSION, null);
-        final String versionType = pluginSetting.getStringOrDefault(DOCUMENT_VERSION_TYPE, null);
-        final boolean normalizeIndex = pluginSetting.getBooleanOrDefault(NORMALIZE_INDEX, false);
-        builder = builder.withNormalizeIndex(normalizeIndex);
 
+        final String versionExpression = openSearchSinkConfig.getVersionExpression();
         builder = builder.withVersionExpression(versionExpression);
         if (versionExpression != null && (!expressionEvaluator.isValidFormatExpression(versionExpression))) {
             throw new InvalidPluginConfigurationException(
                     String.format("document_version \"%s\" is not a valid format expression.", versionExpression));
         }
 
-        builder = builder.withVersionType(versionType);
-
-        if (Objects.nonNull(documentIdField) && Objects.nonNull(documentId)) {
-            throw new InvalidPluginConfigurationException("Both document_id_field and document_id cannot be used at the same time. It is preferred to only use document_id as document_id_field is deprecated.");
-        }
-
-        if (documentIdField != null) {
-            LOG.warn("document_id_field is deprecated in favor of document_id, and support for document_id_field will be removed in a future major version release.");
-            builder = builder.withDocumentIdField(documentIdField);
-        } else if (documentId != null) {
-            builder = builder.withDocumentId(documentId);
-        }
-
-        final String routingField = pluginSetting.getStringOrDefault(ROUTING_FIELD, null);
-        final String routing = pluginSetting.getStringOrDefault(ROUTING, null);
-        if (routingField != null) {
-            LOG.warn("routing_field is deprecated in favor of routing, and support for routing_field will be removed in a future major version release.");
```

Comment:
As mentioned here. We need to keep `routingField` until major version release.

Reply:
routing Field isn't mentioned in the documentation so I wasn't sure if it was already supposed to be removed